pep8.py

```diff
@@ -1319,6 +1323,14 @@ def check_logical(self):
         indent = first_line[:self.mapping[0][1][2][1]]
         self.previous_indent_level = self.indent_level
         self.indent_level = expand_indent(indent)
+
+        # Is this line inside a global (non-indented) function?
+        def is_not_global_class(line):
```

Comment:
This isn't actually perfect. Take for example:

```py
# pkg/compat.py
"""Compat module for <library>."""
import collections

try:
    from collections import OrderedDict
except ImportError:
    class OrderedDict(collections.MutableMapping):
        # ...

try:
    from urlparse import urlparse
except ImportError:
    from urllib.parse import urlparse
```

```py
# pkg/something.py
from .compat import OrderedDict
#...
```

That's a global class and that line does not start with `class` instead it starts with `    class`.